src/lucky/routeable.cr

```diff
@@ -100,6 +118,18 @@ module Lucky::Routeable
 
       is_root_path = path == ""
       path = "/" if is_root_path
+
+      query_params = {} of String => String
+      {% for param in PARAM_DECLARATIONS %}
+        unless {{ param.var }} == {{ param.value || nil }}
```

Comment:
What do you think about extracting an intermediate variable to make it clearer what is going on with this? It took me a bit to get it. Maybe something like this:

```crystal
param_is_default_or_nil = {{ param.var }} == {{ param.value || nil }}
unless param_is_default_or_nil
```